<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        async function fn(){
            let p = new Promise((resolve,reject)=>{
                // resolve('222');
                reject('4444');
            })
            // 如果await右侧为promise对象 那么返回结果就是Promise调用resolve函数所传入的参数
            // let res = await p;
            // console.log(res);

            //如果await右侧是非promise对象 那么返回结果就是它
            // let res = await 222;
            // console.log(res);

            // 如果await右侧是一个返回错误结果的Promise对象 那么需要通过trycatch来捕获
            try {
                let res = await p;
            } catch (e) {
                console.log(e);
            }
        }
        fn()
    </script>
</body>
</html>